toolchain/sem_ir/builtin_function_kind.cpp

```diff
@@ -95,6 +95,27 @@ struct NoReturn {
 // Constraint that a type is `bool`.
 using Bool = BuiltinType<BoolType::TypeInstId>;
 
+// Constraint that a type is `Core.CharLiteral`.
+using CharLiteral = BuiltinType<CharLiteralType::TypeInstId>;
+
+// Constraint that a type is `u8` or an adapted type, including `Core.Char`.
+struct CharCompatible {
+  static auto Check(const File& sem_ir, ValidateState& /*state*/,
+                    TypeId type_id) -> bool {
+    auto int_info = sem_ir.types().TryGetIntTypeInfo(type_id);
+    if (!int_info) {
+      // Not an integer.
+      return false;
+    }
+    if (!int_info->bit_width.has_value() || int_info->is_signed) {
+      // Must be unsigned.
+      return false;
+    }
+    auto bit_width = sem_ir.ints().Get(int_info->bit_width).getZExtValue();
+    return bit_width == 8;
```

Comment:
```suggestion
    return sem_ir.ints().Get(int_info->bit_width) == 8;
```
... using [the comparison against `uint64_t`](https://github.com/llvm/llvm-project/blob/04691aae0dd783162919c85aef5b8b143c0ca411/llvm/include/llvm/ADT/APInt.h#L1069). This is shorter and also won't crash if the bit width somehow doesn't fit in 64 bits :)